---
int32 num_segments
SegmentInfo[] sinfo
